packages/components/src/mobile/link-settings/index.native.js

```diff
@@ -174,6 +175,13 @@ function LinkSettings( {
 	}, [ urlInputValue, labelInputValue, linkRelInputValue, setAttributes ] );
 
 	const onCloseSettingsSheet = useCallback( () => {
+		if ( onCloseSettingsSheetConsumed.current ) {
+			onCloseSettingsSheetConsumed.current = false;
+			return;
+		}
```

Comment:
I think this approach might prevent the logic of `onCloseSettingsSheet` to be executed if the handler is only called once, which is the common behavior. I was thinking about a different approach that resets this value when the `isVisible` props changes from `false` to `true`:

```diff
diff --git a/packages/components/src/mobile/link-settings/index.native.js b/packages/components/src/mobile/link-settings/index.native.js
index f373f54450317e532758d5d277e155bc88e94001..0abebbdeb3fad62cc8a873ff081c97242cafdabe 100644
--- a/packages/components/src/mobile/link-settings/index.native.js
+++ b/packages/components/src/mobile/link-settings/index.native.js
@@ -124,6 +124,10 @@ function LinkSettings( {
 
 	useEffect( () => {
 		const isSettingSheetOpen = isVisible || editorSidebarOpened;
+		if ( isSettingSheetOpen ) {
+			onCloseSettingsSheetConsumed.current = false;
+		}
+
 		if ( options.url.autoFill && isSettingSheetOpen && ! url ) {
 			getURLFromClipboard();
 		}
@@ -176,7 +180,6 @@ function LinkSettings( {
 
 	const onCloseSettingsSheet = useCallback( () => {
 		if ( onCloseSettingsSheetConsumed.current ) {
-			onCloseSettingsSheetConsumed.current = false;
 			return;
 		}
 

```

@jd-alexander  I tested the above change in different cases and so far looks like it's working nice, let me know if you could take a look and if you agree, I'll push the changes.

Reply:
The suggestion here works well @fluiddot I tested it and pushed the changes. 💯

Reply:
Thank you very much @jd-alexander for testing and pushing the changes 🥇 , I'll do a final review and merge it 🎊 .